### ISSUE TITLE
PS 7.3.0 remote command "Invoke-Command" 

### ISSUE THREAD

**Issue by kater-erik** [PowerShell/PowerShell#18699]
### Prerequisites

- [X] Write a descriptive title.
- [X] Make sure you are able to repro it on the [latest released version](https://github.com/PowerShell/PowerShell/releases)
- [X] Search the existing issues.
- [X] Refer to the [FAQ](https://github.com/PowerShell/PowerShell/blob/master/docs/FAQ.md).
- [X] Refer to [Differences between Windows PowerShell 5.1 and PowerShell](https://docs.microsoft.com/powershell/scripting/whats-new/differences-from-windows-powershell).

### Steps to reproduce

Run command "Invoke-Command -Computername XYZ -Scriptblock {Get-Service spooler}" to get the information about the status of the spooler service on a remote system running W2016 server and PSVersion 7.3.0.

Uninstalling PSVersion 7.3.0 and installing PSVersion 7.2.4-win-x64 on the same XYZ system fix the issue and the "Invoke-Command" returns the expected result.

### Expected behavior

```console
Invoke-Command -ComputerName XYZ -ScriptBlock {get-service spoo*}

Status   Name               DisplayName                            PSComputerName
------   ----               -----------                            --------------
Running  Spooler            Print Spooler                          XYZ
```

### Actual behavior

```console
OpenError: [XYZ] Connecting to remote server XYZ failed with the following error message : Could not load file or assembly 'System.Security.Cryptography, Version=7.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'. The system cannot find the file specified. For more information, see the about_Remote_Troubleshooting Help topic.
```

### Error details

```console
OpenError: [XYZ] Connecting to remote server XYZ failed with the following error message : Could not load file or assembly 'System.Security.Cryptography, Version=7.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'. The system cannot find the file specified. For more information, see the about_Remote_Troubleshooting Help topic.
```

### Environment data

```powershell
Name                           Value
----                           -----
PSVersion                      7.3.0
PSEdition                      Core
GitCommitId                    7.3.0
OS                             Microsoft Windows 10.0.14393
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

### Visuals
![PS_error](https://user-images.githubusercontent.com/119589694/205083039-642069a2-dbd5-40a8-99bd-c0a2e888636e.png)

**Comment by dwtaber** [comment]
I'm getting the same behavior from `Enter-PSSession`.

**Comment by kater-erik** [comment]
Invoke-Command -Computername XYZ -Scriptblock {Get-ExecutionPolicy -List}

results in the same behavior:

OpenError: [XYZ] Connecting to remote server XYZ failed with the following error message : Could not load file or assembly 'System.Security.Cryptography, Version=7.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'. The system cannot find the file specified. For more information, see the about_Remote_Troubleshooting Help topic.

So indeed, it does not matter what you put between the {   }

**Comment by dwtaber** [comment]
Why was this issue closed?

**Comment by kater-erik** [comment]
Sorry I am new here and it looks like I closed this issue by accident..

**Comment by JAK1047** [comment]
Same issue here with Enter-PSSession from a Win10 box to a 2022 server. Most I could find .net 7 related was this:

https://github.com/dotnet/aspnetcore/issues/39342

**Comment by JAK1047** [comment]
Confirmed downgrading to v7.2.7 resolves the issue for myself as well.

**Comment by vladhietala** [comment]
I have the same issue, I have one script that runs on a list of servers and servers that have 7.3.0 return this error.

To resolve add `-ConfigurationName "Powershell.7.3.0"` to command. Note that command will fail on servers without 7.3, on these you can use `-ConfigurationName "Powershell.7.2"`

**Comment by JAK1047** [comment]
> I have the same issue, I have one script that runs on a list of servers and servers that have 7.3.0 return this error.
> 
> To resolve add `-ConfigurationName "Powershell.7.3.0"` to command. Note that command will fail on servers without 7.3, on these you can use `-ConfigurationName "Powershell.7.2"`

Going off that I went to check the Session Configurations on the remote box that had also been upgraded and noticed they were all 7.2 only for PSVersion. Trying to create a new PSSessionConfiguration gave:

![image](https://user-images.githubusercontent.com/29665481/206014405-e74c5403-0e80-429e-b5f1-fd995e5de1d5.png)

Reran Enable-PSRemoting and the 7.3 Session Configuration files got created. After that my remoting worked as before.

**Comment by dwtaber** [comment]
Hm.  Running `Enable-PSRemoting` did create a `PowerShell.7.3.0` configuration, but the `PowerShell.7` configuration wasn't updated; it's still looking for PowerShell 7.2, which no longer exists on the machine.

Possible issue with the installer?  I installed via MSI and definitely selected the option to enable remoting.

**Comment by kater-erik** [comment]
> To resolve add `-ConfigurationName "Powershell.7.3.0"` to command. Note that command will fail on servers without 7.3, on these you can use `-ConfigurationName "Powershell.7.2"`

This is a good "work-a-round". Thanks!

**Comment by JAK1047** [comment]
> Hm. Running `Enable-PSRemoting` did create a `PowerShell.7.3.0` configuration, but the `PowerShell.7` configuration wasn't updated; it's still looking for PowerShell 7.2, which no longer exists on the machine.
> 
> Possible issue with the installer? I installed via MSI and definitely selected the option to enable remoting.
> 
> Edit to add: remoting into the new `PowerShell.7.3.0` profile, the handful of basic commands I've tried so far worked without errors, but I'm up for attempting to reproduce errors others encounter.

I also used the MSI and selected that option. The errors I ran into were when attempting to Connect to Microsoft Graph or use the Active Directory module.

![image](https://user-images.githubusercontent.com/29665481/206470524-2d5f3fcb-9160-4117-8319-57351c113d39.png)

Both of course work if I RDP directly to the server or downgrade to 7.2.

**Comment by dwtaber** [comment]
Hm, I'm running into issues even importing the Active Directory module in a remote session, regardless of version—even when the local and remote machine are both using 5.1.  At least some of this is probably due to some quirk of how this domain is configured, but it's complicating my ability to clearly reproduce this issue.

**Comment by kater-erik** [comment]
Just installed PS version 7.3.1 to test, but still need to use the -ConfigurationName "Powershell.7.3.0" in the remote command.

-ConfigurationName "Powershell.7.3.1" does NOT work!!

**Comment by OneScripter** [comment]
I get the same Cryptography error with 7.3.0 installed if I target the "PowerShell.7" configuration.

**Comment by JAK1047** [comment]
At this point I removed all PoSH 7 PSSession Configurations. Completely uninstalled 7.3.1. Rebooted. Reinstalled 7.3.1. Enabled remoting to recreate the Configuration. Connected to the 7.3.1 session. Both the Ms.Graph module from the Powershell Gallery and the Active Directory module from the RSAT server feature give me missing assembly errors. Uninstalled again. Reinstalled 7.2.8. Enabled remoting. Connected to the 7.2.8 session and it works as intended.

Without anything else I can think of to troubleshoot I'm just going to call 7.3 a wash for remoting in my environment until it gets fixed.

**Comment by JAK1047** [comment]
> Hm, I'm running into issues even importing the Active Directory module in a remote session, regardless of version—even when the local and remote machine are both using 5.1. At least some of this is probably due to some quirk of how this domain is configured, but it's complicating my ability to clearly reproduce this issue.

That might just be a double hop issue if you're getting something like "Unable to contact the server". I either have to specify a RunAsAccount in my PSConfiguration or make a Credential object in the session to implicitly pass to the ADCmdlets.

**Comment by OneScripter** [comment]
I rolled everything back to 7.2.8 for now and things are working fine.

**Comment by Ronzan** [comment]
I can confirm, after upgrading to 7.3.1, (re)Enable-PSRemote and could connect but have to use the explicit configuration name, -ConfigurationName "Powershell.7.3.1,
Downgrading the remote servers to 7.2.8 fixed the issue, I can remote from 7.3.1 to 7.2.8 using -ConfigurationName "Powershell.7"

**Comment by kater-erik** [comment]
No I have not tested the PS 7.4 preview, but installing PS 7.3.2 will result that you need to add the explicit configuration name 
(eg -ConfigurationName "Powershell.7.3.2" to your remote command!

**Comment by JAK1047** [comment]
Does just using -ConfigurationName "Powershell.7.3" not work? I would have assumed it would keep getting updated to use the latest along with the version specific Configurations.

Sidenote I did find launching pwsh again from within the remote session after connecting allowed me to use my modules without issue in 7.3.1. No idea why, but it works.

**Comment by kater-erik** [comment]
_Does just using -ConfigurationName "Powershell.7.3" not work?_ 

> sorry a big NO. When target has PS 7.3.2 installed you need to add explicitly "-ConfigurationName "Powershell.7.3.2" to your remote command

**Comment by daxian-dbw** [comment]
This is a duplicate of https://github.com/PowerShell/PowerShell/issues/18587. The root cause was called out in https://github.com/PowerShell/PowerShell/issues/18587#issuecomment-1324375618. The fix will be included in the next 7.3 servicing release 7.3.3. I will keep this issue open till we ship 7.3.3.
